components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/OperationPoliciesApiServiceImpl.java

```diff
@@ -204,27 +222,70 @@ public Response deleteCommonOperationPolicyByPolicyId(String operationPolicyId,
      */
     @Override
     public Response getAllCommonOperationPolicies(Integer limit, Integer offset, String query,
-                                                  MessageContext messageContext) throws APIManagementException {
+            MessageContext messageContext) throws APIManagementException {
 
+        String apiManagementExceptionErrorMessage = "";
+        OperationPolicyDataListDTO policyListDTO = null;
+        String name = null, version = null;
         try {
-            limit = limit != null ? limit : RestApiConstants.PAGINATION_LIMIT_DEFAULT;
-            offset = offset != null ? offset : RestApiConstants.PAGINATION_OFFSET_DEFAULT;
-
-            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
             String organization = RestApiUtil.getValidatedOrganization(messageContext);
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+
+            // If name & version are given, it returns the policy data
+            if (query != null) {
+                Map<String, String> queryParamMap = new HashMap();
+
+                String[] queryParams = query.split(" ");
+                for (String param : queryParams) {
+                    String[] keyVal = param.split(":");
+                    queryParamMap.put(keyVal[0], keyVal[1]);
```

Comment:
we need to check for the array size. otherwise, this will return ArrayIndexOutOfBoundsException